src/ifk_scb_compilations/objects/passenger_transport.py

```diff
@@ -0,0 +1,127 @@
+"""Inputs for request, and analysis."""
+
+from dataclasses import dataclass
+
+import matplotlib.pyplot as plt
+import pandas as pd
+
+
+@dataclass
+class Request_input:
+    """Dataclass for passenger transport query.
+
+    Query info can be found at url.
+    """
+
+    url = "https://api.scb.se/OV0104/v1/doris/sv/ssd/START/MI/MI0107/TotaltUtslappN"
+    query = {
+        "query": [
+            {
+                "code": "Vaxthusgaser",
+                "selection": {"filter": "item", "values": ["CO2-ekv."]},
+            },
+            {
+                "code": "Sektor",
+                "selection": {"filter": "item", "values": ["0.2", "0.4", "8.0", "5.0"]},
+            },
+        ],
+        "response": {"format": "json"},
+    }
+
+
+class Analysis:
+    """Container for plotting data corresponding to fetch spec by Request_input."""
+
+    def __init__(self, request_output: dict) -> None:
+        """Initialization.
+
+        Args:
+            request_output (dict): Output from scb api response.
+        """
+        self.data = self.transform_json_to_df(request_output)
+
+    def transform_json_to_df(self, request_output: dict) -> pd.DataFrame:
+        """Transform json format to dataframe.
+
+        Args:
+            request_output (dict): The first parameter.
+
+        Returns:
+            pd.Dataframe: request output as dataframe
+        """
+        n_vals = len(request_output["data"])
+
+        data_dict = {
+            "emission measure": list(
+                request_output["data"][i]["key"][0] for i in range(n_vals)
+            ),
+            "emission type": list(
+                request_output["data"][i]["key"][1] for i in range(n_vals)
+            ),
+            "year": list(
+                float(request_output["data"][i]["key"][2]) for i in range(n_vals)
+            ),
+            "value": list(
+                float(request_output["data"][i]["values"][0]) for i in range(n_vals)
+            ),
```

Comment:
These list comprehensions can be written using list literals, i.e. 

```python
[request_output["data"][i]["key"][0] for i in range(n_vals)]
```

Reply:
updated.